### ISSUE TITLE
 too many arguments in call to ytypes.Unmarshal

### ISSUE THREAD

**Issue by adibrastegarnia** [google/gnxi#70]
@larryleguo @samribeiro 
Hi
When we try to build the gnmi project and gnmi target  (I need gnmi target which is dependent to gnmi package) using Go modules or dep, we get an error like this: 

pkg/gnmi/modeldata/gostruct/generated.go:71:63: undefined: ytypes.UnmarshalOpt
pkg/gnmi/modeldata/gostruct/generated.go:81:25: too many arguments in call to ytypes.Unmarshal

Any ideas? It looks like there are some conflicts somewhere.

**Comment by adibrastegarnia** [comment]
@larryleguo 

To give you a better idea, I can build it if I don't use any dep management tool. I mean If use dep, I should remove vendor directory and then I can build the project. If I use go modules, I should disable it and then build.

**Comment by adibrastegarnia** [comment]
and list of dependencies if that helps: 

```
require (
        github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
        github.com/golang/protobuf v1.3.1
        github.com/google/gnxi v0.0.0-20190228205329-8521faedac37
        github.com/kylelemons/godebug v1.1.0 // indirect
        github.com/openconfig/gnmi v0.0.0-20180912164834-33a1865c3029
        github.com/openconfig/goyang v0.0.0-20190408185115-e8b0ed2cbb0c
        github.com/openconfig/ygot v0.5.0
        github.com/pmezard/go-difflib v1.0.0 // indirect
        golang.org/x/net v0.0.0-20190522155817-f3200d17e092
        golang.org/x/sys v0.0.0-20190531175056-4c3a928424d2 // indirect
        golang.org/x/text v0.3.2 // indirect
        google.golang.org/genproto v0.0.0-20190530194941-fb225487d101
        google.golang.org/grpc v1.21.0
)
```

**Comment by adibrastegarnia** [comment]
I think the problem is that it retrieves version 0.5.0 which is old (2017).

**Comment by adibrastegarnia** [comment]
I found a solution for that to get master branch using go get and update go.mod. I think ygot should release a new version to address the problem properly.

**Comment by larryleguo** [comment]
Hi, Adib,

I cannot reproduce the issue you have. Can you send me sample code to reproduce it? 

BTW, the error you found:
pkg/gnmi/modeldata/gostruct/generated.go:71:63: undefined: ytypes.UnmarshalOpt
pkg/gnmi/modeldata/gostruct/generated.go:81:25: too many arguments in call to ytypes.Unmarshal
do not match the code in github:
https://github.com/google/gnxi/blob/master/gnmi/modeldata/gostruct/generated.go#L71-L81

**Comment by adibrastegarnia** [comment]
@larryleguo 
Hi 

If you use "go get" directly it works fine because it fetches the master branch but if you use dep or go modules it gets the released version (v0.5.0) which has an old prototype for that function. I fixed it by updating the go.mod file to retrieve the master branch (go get github.com/ygot@master) (look at the go.mod). You can look at the code here: 

https://github.com/onosproject/simulators

**Comment by larryleguo** [comment]
gnxi does not have a released version so far. The released version v0.5.0 you mentioned is the openconfig ygot. I think your vendor directory was not updated, causing the problem. I see Rob has replied https://github.com/openconfig/ygot/issues/287, and the v0.6.0 is just released (in https://github.com/openconfig/ygot/releases). Close this issue now.
